src/mpid/ch4/src/ch4_startall.h

```diff
@@ -28,18 +28,7 @@ MPL_STATIC_INLINE_PREFIX int MPID_Startall(int count, MPIR_Request * requests[])
     int i;
     for (i = 0; i < count; i++) {
         MPIR_Request *req = requests[i];
-        /* This is sub-optimal, can we do better? */
-        if (MPIDI_CH4I_REQUEST_ANYSOURCE_PARTNER(req)) {
-            mpi_errno = MPIDI_SHM_mpi_startall(1, &req);
-            if (mpi_errno == MPI_SUCCESS) {
-                mpi_errno = MPIDI_NM_mpi_startall(1, &MPIDI_CH4I_REQUEST_ANYSOURCE_PARTNER(req));
-                MPIDI_CH4I_REQUEST_ANYSOURCE_PARTNER(req->u.persist.real_request) =
-                    MPIDI_CH4I_REQUEST_ANYSOURCE_PARTNER(req)->u.persist.real_request;
-                MPIDI_CH4I_REQUEST_ANYSOURCE_PARTNER(MPIDI_CH4I_REQUEST_ANYSOURCE_PARTNER
-                                                     (req)->u.persist.real_request) =
-                    req->u.persist.real_request;
-            }
-        } else if (MPIDI_CH4I_REQUEST(req, is_local))
+        if (MPIDI_CH4I_REQUEST(req, is_local))
```

Comment:
Is the change here meant to prevent double launching? Then we don't need the fix in `ch4_recv.h`, right?